xbmc/cores/AudioEngine/Utils/AEUtil.cpp

```diff
@@ -607,3 +607,25 @@ int CAEUtil::GetAVChannelIndex(enum AEChannel aechannel, uint64_t layout)
   av_channel_layout_uninit(&ch_layout);
   return idx;
 }
+
+void CAEUtil::GenerateSilence(AEDataFormat format,
+                              unsigned int frameSize,
+                              void* buffer,
+                              unsigned int frames)
+
+{
+  const unsigned int dataLength{frames * frameSize};
+
+  switch (format)
+  {
+    case AV_SAMPLE_FMT_U8:
+    case AV_SAMPLE_FMT_U8P:
+      memset(buffer, 0x80, dataLength);
```

Comment:
For Signed formats I would agree with you, but this is Unsigned formats. Where is the 0x80 coming from? Shouldn't it be 0x00 as well?

Reply:
And surprisingly audioconvert.c in ffmpeg does the same.

Reply:
Nice that's historically ... they also use it signed ... why the format is name U8 ... funny ... all good.

Reply:
I wasn't sure between 0x80 and 0x7F due to lack of samples, thanks for the confirmation.
The value is due to the nature of audio data, for a good signal silence is the middle of the range of possible values.
Unsigned  byte: 0 to 0xFF, the middle value is 0x80 (or 0x7F...).
For signed types, the middle of the range is 0.